src/interpreter/mod.rs

```diff
@@ -129,9 +129,9 @@ enum TerminatorTarget {
 }
 
 #[derive(Clone, Debug, Eq, PartialEq, Hash)]
-enum ConstantId {
+enum ConstantId<'tcx> {
     Promoted { index: usize },
-    Static { def_id: DefId },
+    Static { def_id: DefId, substs: &'tcx Substs<'tcx> },
```

Comment:
Good point. Statics can't have substs since they aren't allowed to use the function-level type parameters.